Branch name based version doesn't work

I've heard pkg.pr.new supports a branch name as version such as https://pkg.pr.new/svelte@async in https://github.com/sveltejs/svelte/discussions/15845. However when I tried something similar in https://github.com/vitejs/vite-plugin-react/pull/634 to have https://pkg.pr.new/@vitejs/plugin-rsc@canary, it ended up 404. It seems to only work with exact commit number https://pkg.pr.new/@vitejs/plugin-rsc@950ed88 or pr number https://pkg.pr.new/@vitejs/plugin-rsc@634. 

```
 ERR_PNPM_FETCH_404  GET https://pkg.pr.new/@vitejs/plugin-rsc@canary: Not Found - 404

This error happened while installing a direct dependency of /home/hiroshi/code/others/waku

No authorization header was set for the request.
```

--- a/packages/app/server/routes/publish.post.ts
+++ b/packages/app/server/routes/publish.post.ts
@@ -202,6 +202,20 @@ export default eventHandler(async (event) => {
       });
     }
 
+    if (
+      workflowData.headBranch &&
+      workflowData.headBranch !== workflowData.ref
+    ) {
+      const branchCursorKey = `${baseKey}:${workflowData.headBranch}`;
+      const branchCursor = await cursorBucket.getItem(branchCursorKey);
+      if (!branchCursor || branchCursor.timestamp < runId) {
+        await cursorBucket.setItem(branchCursorKey, {
+          sha: workflowData.sha,
+          timestamp: runId,
+        });
+      }
+    }
+
     await workflowsBucket.removeItem(key);
 
     const urls = packagesWithoutPrefix.map((packageName) =>

--- a/packages/app/server/routes/webhook.post.ts
+++ b/packages/app/server/routes/webhook.post.ts
@@ -55,11 +55,18 @@ export default eventHandler(async (event) => {
       const lookupKey = isNewPullRequest ? prKey : oldPrDataHash;
       const prNumber = await pullRequestNumbersBucket.getItem(lookupKey);
 
+      const headBranch = payload.workflow_run.head_branch;
+      const isHeadOnUpstream =
+        payload.workflow_run.head_repository?.full_name ===
+        payload.repository.full_name;
+
       const data: WorkflowData = {
         owner,
         repo,
         sha: payload.workflow_run.head_sha,
-        ref: isPullRequest ? `${prNumber}` : payload.workflow_run.head_branch!, // it's a pull request workflow
+        ref: isPullRequest ? `${prNumber}` : headBranch!,
+        headBranch:
+          isPullRequest && isHeadOnUpstream && headBranch ? headBranch : null,
       };
 
       // Publishing is only available throughout the lifetime of a workflow_job

--- a/packages/app/server/types.ts
+++ b/packages/app/server/types.ts
@@ -3,6 +3,7 @@ export interface WorkflowData {
   repo: string;
   sha: string;
   ref: string;
+  headBranch?: string | null;
 }
 
 export interface PullRequestData {

